### PR TITLE
remove microsoft.aspnetcore.authentication.oauth dependency from netcoreapp3.0 TFM

### LINKED ISSUE #217: Remove Microsoft.AspNetCore.Authorization.OAuth for netcore3x?
My (limited) understanding is that this is now a framework reference and so perhaps isn't required as a nuget package?

### PR DESCRIPTION
**Pull request by baronfel**
This dependency is in the framework ref now and so no longer should be included. All remaining 2.2 references come from the azure functions SDK reference, which I haven't figured out how to remove.

The other netcoreapp3.0 references are ones that build on top of the core primitives that are in the aspnetcore framework ref, and do still exist on nuget.

Fixes #217.

### CODE REVIEW

File: paket.dependencies [hunk 1]
```diff
@@ -5,7 +5,6 @@ storage: none
 framework: netstandard2.0, netcoreapp3.0
 
 nuget Expecto
-nuget Microsoft.NET.Sdk.Functions >= 1.0.24
```
Comment (baronfel) [inline paket.dependencies]: we can remove this because there's a matching functions sdk ref in each of the netstandard2.0 and netcoreapp3.0 groups.

**Comment by isaacabraham** [comment]
Can't the authorisation one also go?

**Comment by baronfel** [comment]
@isaacabraham I don't believe so, that one is still distributed over nuget for netcoreapp3.1 from what I can see: https://www.nuget.org/packages/Microsoft.AspNetCore.Authorization/

**Comment by isaacabraham** [comment]
@baronfel see https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.1&tabs=visual-studio#add-package-references-for-removed-assemblies - the screenshot suggests that Authorization is also still going to be a framework ref.

**Comment by baronfel** [comment]
Yes, you're right with this. I still hate that this is the mechanism that we have to determine what's in the box :-/

For later reference, here's the whole set of included dlls:

<details>
<summary> in the box dlls </summary>

```
Microsoft.AspNetCore.Antiforgery.dll
Microsoft.AspNetCore.App.deps.json
Microsoft.AspNetCore.App.runtimeconfig.json
Microsoft.AspNetCore.Authentication.Abstractions.dll
Microsoft.AspNetCore.Authentication.Cookies.dll
Microsoft.AspNetCore.Authentication.Core.dll
Microsoft.AspNetCore.Authentication.OAuth.dll
Microsoft.AspNetCore.Authentication.dll
Microsoft.AspNetCore.Authorization.Policy.dll
Microsoft.AspNetCore.Authorization.dll
Microsoft.AspNetCore.Components.Authorization.dll
Microsoft.AspNetCore.Components.Forms.dll
Microsoft.AspNetCore.Components.Server.dll
Microsoft.AspNetCore.Components.Web.dll
Microsoft.AspNetCore.Components.dll
Microsoft.AspNetCore.Connections.Abstractions.dll
Microsoft.AspNetCore.CookiePolicy.dll
Microsoft.AspNetCore.Cors.dll
Microsoft.AspNetCore.Cryptography.Internal.dll
Microsoft.AspNetCore.Cryptography.KeyDerivation.dll
Microsoft.AspNetCore.DataProtection.Abstractions.dll
Microsoft.AspNetCore.DataProtection.Extensions.dll
Microsoft.AspNetCore.DataProtection.dll
Microsoft.AspNetCore.Diagnostics.Abstractions.dll
Microsoft.AspNetCore.Diagnostics.HealthChecks.dll
Microsoft.AspNetCore.Diagnostics.dll
Microsoft.AspNetCore.HostFiltering.dll
Microsoft.AspNetCore.Hosting.Abstractions.dll
Microsoft.AspNetCore.Hosting.Server.Abstractions.dll
Microsoft.AspNetCore.Hosting.dll
Microsoft.AspNetCore.Html.Abstractions.dll
Microsoft.AspNetCore.Http.Abstractions.dll
Microsoft.AspNetCore.Http.Connections.Common.dll
Microsoft.AspNetCore.Http.Connections.dll
Microsoft.AspNetCore.Http.Extensions.dll
Microsoft.AspNetCore.Http.Features.dll
Microsoft.AspNetCore.Http.dll
Microsoft.AspNetCore.HttpOverrides.dll
Microsoft.AspNetCore.HttpsPolicy.dll
Microsoft.AspNetCore.Identity.dll
Microsoft.AspNetCore.Localization.Routing.dll
Microsoft.AspNetCore.Localization.dll
Microsoft.AspNetCore.Metadata.dll
Microsoft.AspNetCore.Mvc.Abstractions.dll
Microsoft.AspNetCore.Mvc.ApiExplorer.dll
Microsoft.AspNetCore.Mvc.Core.dll
Microsoft.AspNetCore.Mvc.Cors.dll
Microsoft.AspNetCore.Mvc.DataAnnotations.dll
Microsoft.AspNetCore.Mvc.Formatters.Json.dll
Microsoft.AspNetCore.Mvc.Formatters.Xml.dll
Microsoft.AspNetCore.Mvc.Localization.dll
Microsoft.AspNetCore.Mvc.Razor.dll
Microsoft.AspNetCore.Mvc.RazorPages.dll
Microsoft.AspNetCore.Mvc.TagHelpers.dll
Microsoft.AspNetCore.Mvc.ViewFeatures.dll
Microsoft.AspNetCore.Mvc.dll
Microsoft.AspNetCore.Razor.Runtime.dll
Microsoft.AspNetCore.Razor.dll
Microsoft.AspNetCore.ResponseCaching.Abstractions.dll
Microsoft.AspNetCore.ResponseCaching.dll
Microsoft.AspNetCore.ResponseCompression.dll
Microsoft.AspNetCore.Rewrite.dll
Microsoft.AspNetCore.Routing.Abstractions.dll
Microsoft.AspNetCore.Routing.dll
Microsoft.AspNetCore.Server.HttpSys.dll
Microsoft.AspNetCore.Server.IIS.dll
Microsoft.AspNetCore.Server.IISIntegration.dll
Microsoft.AspNetCore.Server.Kestrel.Core.dll
Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.dll
Microsoft.AspNetCore.Server.Kestrel.dll
Microsoft.AspNetCore.Session.dll
Microsoft.AspNetCore.SignalR.Common.dll
Microsoft.AspNetCore.SignalR.Core.dll
Microsoft.AspNetCore.SignalR.Protocols.Json.dll
Microsoft.AspNetCore.SignalR.dll
Microsoft.AspNetCore.StaticFiles.dll
Microsoft.AspNetCore.WebSockets.dll
Microsoft.AspNetCore.WebUtilities.dll
Microsoft.AspNetCore.dll
Microsoft.Extensions.Caching.Abstractions.dll
Microsoft.Extensions.Caching.Memory.dll
Microsoft.Extensions.Configuration.Abstractions.dll
Microsoft.Extensions.Configuration.Binder.dll
Microsoft.Extensions.Configuration.CommandLine.dll
Microsoft.Extensions.Configuration.EnvironmentVariables.dll
Microsoft.Extensions.Configuration.FileExtensions.dll
Microsoft.Extensions.Configuration.Ini.dll
Microsoft.Extensions.Configuration.Json.dll
Microsoft.Extensions.Configuration.KeyPerFile.dll
Microsoft.Extensions.Configuration.UserSecrets.dll
Microsoft.Extensions.Configuration.Xml.dll
Microsoft.Extensions.Configuration.dll
Microsoft.Extensions.DependencyInjection.Abstractions.dll
Microsoft.Extensions.DependencyInjection.dll
Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions.dll
Microsoft.Extensions.Diagnostics.HealthChecks.dll
Microsoft.Extensions.FileProviders.Abstractions.dll
Microsoft.Extensions.FileProviders.Composite.dll
Microsoft.Extensions.FileProviders.Embedded.dll
Microsoft.Extensions.FileProviders.Physical.dll
Microsoft.Extensions.FileSystemGlobbing.dll
Microsoft.Extensions.Hosting.Abstractions.dll
Microsoft.Extensions.Hosting.dll
Microsoft.Extensions.Http.dll
Microsoft.Extensions.Identity.Core.dll
Microsoft.Extensions.Identity.Stores.dll
Microsoft.Extensions.Localization.Abstractions.dll
Microsoft.Extensions.Localization.dll
Microsoft.Extensions.Logging.Abstractions.dll
Microsoft.Extensions.Logging.Configuration.dll
Microsoft.Extensions.Logging.Console.dll
Microsoft.Extensions.Logging.Debug.dll
Microsoft.Extensions.Logging.EventLog.dll
Microsoft.Extensions.Logging.EventSource.dll
Microsoft.Extensions.Logging.TraceSource.dll
Microsoft.Extensions.Logging.dll
Microsoft.Extensions.ObjectPool.dll
Microsoft.Extensions.Options.ConfigurationExtensions.dll
Microsoft.Extensions.Options.DataAnnotations.dll
Microsoft.Extensions.Options.dll
Microsoft.Extensions.Primitives.dll
Microsoft.Extensions.WebEncoders.dll
Microsoft.JSInterop.dll
Microsoft.Net.Http.Headers.dll
Microsoft.Win32.SystemEvents.dll
System.Diagnostics.EventLog.dll
System.Drawing.Common.dll
System.IO.Pipelines.dll
System.Security.Cryptography.Pkcs.dll
System.Security.Cryptography.Xml.dll
System.Security.Permissions.dll
System.Windows.Extensions.dll
THIRD-PARTY-NOTICES.txt
```
</details>

**Comment by baronfel** [comment]
@isaacabraham want to take a look after this latest commit? turns out both authorization and caching are in the framework so both could be removed.

**Comment by isaacabraham** [comment]
LGTM

**Comment by isaacabraham** [comment]
@baronfel something doesn't look quite right: see https://www.nuget.org/packages/Saturn/

image

OAuth, Authorization and Caching Memory are still there (unless this PR hasn't yet been released @Krzysztof-Cieslak?)

**Comment by Krzysztof-Cieslak** [comment]
Yes, I haven’t released it yet

**Pull request merged**